coverage.txt

```diff
@@ -1,11 +1,11 @@
 Name                          Stmts   Miss  Cover
 -------------------------------------------------
 srnca/__init__.py                 0      0   100%
-srnca/nca.py                     75      0   100%
-srnca/utils.py                   65      0   100%
+srnca/nca.py                    112      0   100%
+srnca/utils.py                   69      0   100%
 testing/__init__.py               0      0   100%
-testing/srnca/test_nca.py        31      0   100%
+testing/srnca/test_nca.py        70      0   100%
```

Comment:
Note I used a single-line conditional trick to get full line coverage testing to_device(_) even  when no cuda is available.

```
print("") if torch.cuda.is_available() else \
                    print(f"warning, cuda not found but {my_device} specified, falling back to cpu")
self.my_device = torch.device(my_device) if torch.cuda.is_available() else torch.device("cpu")
```